tensorpipe/test/channel/cuda_ipc/cuda_ipc_test.cc

```diff
@@ -17,70 +17,23 @@
 
 namespace {
 
-class CudaWrapper : public DataWrapper {
+class CudaChannelTestHelper : public ChannelTestHelper<tensorpipe::CudaTensor> {
  public:
-  explicit CudaWrapper(size_t len) : size_(len) {
-    EXPECT_EQ(cudaSuccess, cudaSetDevice(0));
-    if (size_ > 0) {
-      EXPECT_EQ(cudaSuccess, cudaMalloc(&cudaData_, size_));
-    }
-  }
-
-  void* data() override {
-    return cudaData_;
+  std::shared_ptr<tensorpipe::channel::CudaContext> makeContext(
+      std::string id) override {
+    auto context = std::make_shared<tensorpipe::channel::cuda_ipc::Context>();
+    context->setId(std::move(id));
+    return context;
   }
 
-  size_t size() override {
-    return size_;
+  std::shared_ptr<PeerGroup> makePeerGroup() override {
+    return std::make_shared<ProcessPeerGroup>();
   }
-
-  void wrap(const void* ptr) override {
-    if (size_ > 0) {
-      EXPECT_EQ(
-          cudaSuccess, cudaMemcpy(cudaData_, ptr, size_, cudaMemcpyDefault));
-    }
-  }
-
-  void unwrap(void* ptr) override {
-    if (size_ > 0) {
-      EXPECT_EQ(
-          cudaSuccess, cudaMemcpy(ptr, cudaData_, size_, cudaMemcpyDefault));
-    }
-  }
-
-  ~CudaWrapper() override {
-    if (size_ > 0) {
-      EXPECT_EQ(cudaSuccess, cudaFree(cudaData_));
-    }
-  }
-
- private:
-  void* cudaData_;
-  size_t size_;
 };
 
-// class CudaChannelTestHelper : public ChannelTestHelper {
-//  public:
-//   std::shared_ptr<tensorpipe::channel::Context> makeContext(
-//       std::string id) override {
-//     auto context =
-//     std::make_shared<tensorpipe::channel::cuda_ipc::Context>();
-//     context->setId(std::move(id));
-//     return context;
-//   }
+CudaChannelTestHelper helper;
 
-//   std::shared_ptr<PeerGroup> makePeerGroup() override {
-//     return std::make_shared<ProcessPeerGroup>();
-//   }
-
-//   std::shared_ptr<DataWrapper> makeBuffer(size_t len) override {
-//     return std::make_shared<CudaWrapper>(len);
-//   }
-// };
-
-// CudaChannelTestHelper helper;
-
-class CudaIpcChannelTest : public ChannelTest {};
+class CudaIpcChannelTest : public CudaChannelTest {};
```

Comment:
What is this doing exactly? I don't see anything equivalent in the other channels.